tests/nosetests/pyanaconda_tests/module_payload_live_os_test.py

```diff
@@ -42,25 +44,100 @@ def setUp(self):
         self.live_os_module = LiveOSHandlerModule()
         self.live_os_interface = LiveOSHandlerInterface(self.live_os_module)
 
+        self.source_tests = SourceSharedTest(self,
+                                             payload=self.live_os_module,
+                                             payload_intf=self.live_os_interface)
+
         self.callback = Mock()
         self.live_os_interface.PropertiesChanged.connect(self.callback)
 
-    def _prepare_and_use_source(self):
-        source = create_autospec(LiveOSSourceModule())
-        source.image_path = "/test/path"
-        source.type = SourceType.LIVE_OS_IMAGE
-        source.is_ready.return_value = True
+    def _prepare_source(self):
+        return self.source_tests.prepare_source(SourceType.LIVE_OS_IMAGE)
 
-        self.live_os_module.set_sources([source])
+    def _prepare_and_use_source(self):
+        source = self._prepare_source()
+        self.source_tests.set_sources([source])
 
         return source
 
+    def supported_sources_test(self):
+        """Test LiveOS supported sources API."""
+        self.assertEqual(
+            [SourceType.LIVE_OS_IMAGE.value],
+            self.live_os_interface.SupportedSourceTypes)
+
+    def sources_empty_test(self):
+        """Test sources LiveOS API for emptiness."""
+        self.source_tests.check_empty_sources()
+
+    @patch("pyanaconda.modules.payload.payloads.payload_base_interface.PayloadSourceContainer")
+    def set_source_test(self, source_container):
+        """Test if set source API of LiveOS payload."""
+        paths = {
+            "Source1": self._prepare_source()
+        }
+
+        self.source_tests.check_set_sources(paths, source_container)
+
+    @patch("pyanaconda.modules.payload.payloads.payload_base_interface.PayloadSourceContainer")
+    def set_multiple_sources_fail_test(self, source_container):
+        """Test LiveOS payload can't set multiple sources."""
+        paths = {
+            "Source1": self._prepare_source(),
+            "Source2": self._prepare_source()
+        }
+
+        self.source_tests.check_set_sources(paths,
+                                            source_container,
+                                            exception=IncompatibleSourceError)
+
+    @patch("pyanaconda.modules.payload.payloads.payload_base_interface.PayloadSourceContainer")
+    def set_when_initialized_source_fail_test(self, source_container):
+        """Test LiveOS payload can't set new sources if the old ones are initialized."""
+        source1 = self._prepare_source()
+        source2 = self._prepare_source()
+
+        paths = {
+            "Source1": source1
+        }
+        paths2 = {
+            "Source2": source2
+        }
+
+        set_up_dbus_container_mock(source_container, paths)
```

Comment:
This is really not necessary. If you patch only `DBus.publish_object`, you can call `PayloadSourceContainer.to_object_path` directly to create the mapping and get unique DBus paths for your source objects. Then `SetSources` will work as expected.

Reply:
Fixed.

Reply:
Fixed.